src/ResourceManagement/PrivateDns/Domain/IPrivateDnsZones.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.PrivateDns.Fluent
+{
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core.CollectionActions;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Entry point to private DNS zone management API in Azure.
+    /// </summary>
+    public interface IPrivateDnsZones :
+        ISupportsCreating<PrivateDnsZone.Definition.IBlank>,
+        ISupportsGettingById<IPrivateDnsZone>,
+        ISupportsGettingByResourceGroup<IPrivateDnsZone>
+    {
+        /// <summary>
+        /// Deletes a private zone from Azure, identifying it by its resource ID.
+        /// </summary>
+        /// <param name="id">The resource ID of the private zone to delete.</param>
+        /// <param name="eTagValue">The ETag value to set on IfMatch header for concurrency protection.</param>
+        void DeleteById(string id, string eTagValue = default(string));
```

Comment:
If my understanding is correct, ETag is not a property of any particluar resource, but a generic property used in  REST API mainly to avoid concurrent edit of same resource.
https://www.ibm.com/support/knowledgecenter/SSQP76_8.9.0/com.ibm.odm.itoa.admin/topics/tsk_entities_update_etag.html

A question which I might not have the answer right now:

Should the ETag used e.g. by the value the UPDATE / DELETE operation populated automatically from ETag of the particular resource.

Reply:
Nice catch:) Actually, there are some discussion regarding etag from here: https://github.com/Azure/azure-sdk/pull/646  Could you help extract the information from here and document them for SDK